awscli/compat.py

```diff
@@ -343,4 +344,117 @@ def ignore_user_entered_signals():
         yield
     finally:
         for sig, user_signal in enumerate(signal_list):
-            signal.signal(user_signal, actual_signals[sig])
\ No newline at end of file
+            signal.signal(user_signal, actual_signals[sig])
+
+
+# linux_distribution is used by the CodeDeploy customization. Python 3.8
+# removed it from the stdlib, so it is vendored here in the case where the
+# import fails.
+try:
+    from platform import linux_distribution
+except ImportError:
+    _UNIXCONFDIR = '/etc'
+    _supported_dists = (
+        'SuSE', 'debian', 'fedora', 'redhat', 'centos',
+        'mandrake', 'mandriva', 'rocks', 'slackware', 'yellowdog', 'gentoo',
+        'UnitedLinux', 'turbolinux', 'arch', 'mageia')
+    _release_filename = re.compile(r'(\w+)[-_](release|version)', re.ASCII)
+    def linux_distribution(distname='', version='', id='',
+                           supported_dists=_supported_dists,
+                           full_distribution_name=1):
+        return _linux_distribution(distname, version, id, supported_dists,
+                                   full_distribution_name)
+
+    def _linux_distribution(distname, version, id, supported_dists,
+                            full_distribution_name):
+
+        """ Tries to determine the name of the Linux OS distribution name.
+            The function first looks for a distribution release file in
+            /etc and then reverts to _dist_try_harder() in case no
+            suitable files are found.
+            supported_dists may be given to define the set of Linux
+            distributions to look for. It defaults to a list of currently
+            supported Linux distributions identified by their release file
+            name.
+            If full_distribution_name is true (default), the full
+            distribution read from the OS is returned. Otherwise the short
+            name taken from supported_dists is used.
+            Returns a tuple (distname, version, id) which default to the
+            args given as parameters.
+        """
+        try:
+            etc = os.listdir(_UNIXCONFDIR)
+        except OSError:
+            # Probably not a Unix system
+            return distname, version, id
+        etc.sort()
+        for file in etc:
+            m = _release_filename.match(file)
+            if m is not None:
+                _distname, dummy = m.groups()
+                if _distname in supported_dists:
+                    distname = _distname
+                    break
+        else:
+            return _dist_try_harder(distname, version, id)
+
+        # Read the first line
+        with open(os.path.join(_UNIXCONFDIR, file), 'r',
+                  encoding='utf-8', errors='surrogateescape') as f:
+            firstline = f.readline()
+        _distname, _version, _id = _parse_release_file(firstline)
```

Comment:
Please test this on ubuntu and red hat to confirm we get the same behavior as the stdlib version in earlier pythons.  Right now this will fail because `_parse_release_file` isn't included here.